tests/Type/QueryPlanTest.php

```diff
@@ -298,6 +298,60 @@ public function testQueryPlan() : void
         self::assertFalse($queryPlan->hasType('Test'));
     }
 
+    public function testQueryPlanForWrappedTypes() : void
+    {
+        $article = new ObjectType([
+            'name'   => 'Article',
+            'fields' => [
+                'id' => ['type' => Type::string()],
+            ],
+        ]);
+
+        $doc               = '
+      query Test {
+        articles {
+            id
+        }
+      }
+';
+        $expectedQueryPlan = [
+            'id' => [
+                'type' => Type::string(),
+                'args' => [],
+                'fields' => [],
+            ],
+        ];
+
+        /** @var QueryPlan|null $queryPlan */
+        $queryPlan = null;
+
+        $blogQuery = new ObjectType([
+            'name'   => 'Query',
+            'fields' => [
+                'articles' => [
+                    'type'    => Type::nonNull(Type::listOf($article)),
+                    'resolve' => static function (
+                        $value,
+                        $args,
+                        $context,
+                        ResolveInfo $info
+                    ) use (
+                        &$queryPlan
+                    ) : array {
+                        $queryPlan = $info->lookAhead();
+
+                        return [];
+                    },
+                ],
+            ],
+        ]);
+
+        $schema = new Schema(['query' => $blogQuery]);
+        GraphQL::executeQuery($schema, $doc);
+
+        self::assertEquals($expectedQueryPlan, $queryPlan->queryPlan());
```

Comment:
well mb we should use `assertSame` or another set of assertions. `assertEquals` might incorrectly compare objects which might lead to bugs when test fixtures are modified in the future.

Reply:
okay will change